lib/Doctrine/DBAL/Query/Expression/ExpressionBuilder.php

```diff
@@ -256,7 +259,8 @@ public function isNotNull($x)
      */
     public function like($x, $y)
     {
-        return $this->comparison($x, 'LIKE', $y);
+        return $this->comparison($x, 'LIKE', $y) .
+            (func_num_args() >= 3 ? sprintf(" ESCAPE '%s'", func_get_args()[2]) : '');
```

Comment:
Should I add doc for that new, simulated parameter, or will that confuse CS tools? Should I also add something like  `/* ?$escapeChar = null */` in the signature?

Reply:
Not sure what the generally used approach is. Let's try both the phpDoc parameter and commented out method argument, see how PHPCS reacts.

BTW, shouldn't `ESCAPE '%s'` be `ESCAPE %s`? Otherwise, the caller won't be able to bind the escape character a parameter. It's the platform's responsibility to quote literals, not the expression builder's.

Reply:
Also, do you want to use `func_get_arg(2)` instead of `func_get_args()[2]`?